lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1127,6 +1149,7 @@ struct FIRRTLLowering : public FIRRTLVisitor<FIRRTLLowering, LogicalResult> {
                                  std::function<void(void)> elseCtor = {});
   void addIfProceduralBlock(Value cond, std::function<void(void)> thenCtor,
                             std::function<void(void)> elseCtor = {});
+  Value mapArray(Value array, std::function<Value(Value)> fn);
```

Comment:
I think you can use llvm::unique_function which is a bit more efficient than std::function for this usecase

Reply:
I changed the interface not to pass function pointer. Thanks!